### PR TITLE
add 🐬  functionality

### PR DESCRIPTION
**Pull request by Jutho**
Flipping tensor indices, for @Adwait-Naravane and @VictorVanthilt . 

Be careful with fermions, as shown in the tests, simply flipping both of a pair of contracted indices does not yield the same contraction when fermions are involved.

### CODE REVIEW

**Comment by Jutho** [comment]
Ok this is not correct yet. I will have to investigate further.

File: src/tensors/indexmanipulations.jl [hunk 1]
```diff
@@ -1,5 +1,21 @@
...
+"""
+function flip(t::AbstractTensorMap, I)
+    P = flip(space(t), I)
+    t2 = similar(t, P)
+    for (c, b) in blocks(t)
+        copy!(t2[c], b)
+    end
+    return t
+end
+flip(t::TensorMap, I) = TensorMap(t.data, flip(space(t), I))
```
Comment (lkdvos) [inline src/tensors/indexmanipulations.jl]: I think here it's probably best to also include a `copy` keyword, similar to `permute`, which can be used to control the sharing of data. At least, it's good to have this in the docstring somewhere

**Review by lkdvos — COMMENTED** [review]
Definitely like the ability to have this!

Just as a sidenote, I'm fine if this is not possible or we don't have the time for this yet:
Do you know if there is any way to make this consistently work with fermions? For example, what happens if we define `flip` to be:

- `i` in codomain and `!isdual` -> flip
- `i` in codomain and `isdual` -> flip + twist
- `i` in domain and `!isdual` -> flip + twist
- `i` in domain and `isdual` -> flip
I think In some sense, what I am thinking of is that `contract` would basically become equivalent to `flip`ping all contracted indices to be ``<bra|ket>`` instead of the other way around, and then contracting. I'm just not entirely sure if this is consistent, as in that it commutes with arbitrary permutations + contractions

**Comment by Jutho** [comment]
Ok, this turned out slightly more complicated as anticipated, because of Frobenius Schur factors. However, I think the implementation is now fixed, and includes the suggestion of @lkdvos to also include twists. I think this guarantees that if both spaces in a pair of `@tensor` contracted indices are flipped, then the contraction is unaltered, irrespective of whether either of the indices live in the domain or codomain. This might fail in an `@planar` context however.

**Pull request merged**